provider/lxd/environ_test.go

```diff
@@ -377,36 +377,20 @@ type environProfileSuite struct {
 	lxd.EnvironSuite
 
 	callCtx context.ProviderCallContext
+
+	svr    *lxd.MockServer
+	lxdEnv environs.LXDProfiler
 }
 
 var _ = gc.Suite(&environProfileSuite{})
 
-func (s *environProfileSuite) TestMaybeWriteLXDProfile(c *gc.C) {
-	ctrl := gomock.NewController(c)
-	defer ctrl.Finish()
+func (s *environProfileSuite) TestMaybeWriteLXDProfileYes(c *gc.C) {
+	defer s.setup(c).Finish()
 
-	svr := lxd.NewMockServer(ctrl)
-	exp := svr.EXPECT()
-	gomock.InOrder(
-		exp.HasProfile("testname").Return(true, nil),
-		exp.HasProfile("testname").Return(false, nil),
-		exp.CreateProfile(api.ProfilesPost{
-			Name: "testname",
-			ProfilePut: api.ProfilePut{
-				Config: map[string]string{
-					"security.nesting": "true",
-				},
-				Description: "test profile",
-			},
-		}).Return(nil),
-	)
+	profile := "testname"
+	s.expectMaybeWriteLXDProfile(false, profile)
```

Comment:
some noise in this file, i wanted to update to the pattern of expect*() for setup while here.